app/src/pages/inside/testCaseLibraryPage/tagPopover/tagPopover.tsx

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2025 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ReactNode, useState } from 'react';
+import { useIntl } from 'react-intl';
+import { BubblesLoader, Button, Popover } from '@reportportal/ui-kit';
+import { PopoverProps } from '@reportportal/ui-kit/popover';
+
+import { createClassnames } from 'common/utils';
+import { SearchField } from 'components/fields/searchField';
+
+import { Attribute } from '../types';
+import { useTagSearch, TagError } from './useTagSearch';
+import { messages } from './messages';
+import { commonMessages } from '../commonMessages';
+
+import styles from './tagPopover.scss';
+import { isEmpty } from 'es-toolkit/compat';
+
+const cx = createClassnames(styles);
+
+interface TagPopoverProps extends Pick<PopoverProps, 'placement'> {
+  trigger: ReactNode;
+  onTagSelect: (tag: Attribute) => void;
+  className?: string;
+}
+
+export const TagPopover = ({
+  trigger,
+  onTagSelect,
+  placement = 'bottom-start',
+  className,
+}: TagPopoverProps) => {
+  const { formatMessage } = useIntl();
+  const [searchValue, setSearchValue] = useState<string>('');
+  const [isOpened, setIsOpened] = useState<boolean>(false);
+  const { tags, loading, error, createTag } = useTagSearch(searchValue);
+
+  const handleSearchChange = (value: string) => {
+    setSearchValue(value);
+  };
+
+  const handleTagClick = (tag: Attribute) => {
+    onTagSelect(tag);
+    setSearchValue('');
+    setIsOpened(false);
+  };
+
+  const handleCreateTag = () => {
+    if (!searchValue.trim()) {
```

Comment:
Let's store to variable and use in both places

Reply:
reused